jobs/JGLOBAL_ATMOS_POST

```diff
@@ -1,46 +1,51 @@
 #! /usr/bin/env bash
 
-source "$HOMEgfs/ush/preamble.sh"
+# shellcheck source=ush/preamble.sh
+source "${HOMEgfs:?}/ush/preamble.sh"
 
 export RUN_ENVIR=${RUN_ENVIR:-"nco"}
 
 #############################
 # Source relevant config files
 #############################
 configs="base post"
-export EXPDIR=${EXPDIR:-$HOMEgfs/parm/config}
-config_path=${EXPDIR:-$PACKAGEROOT/gfs.${gfs_ver}/parm/config}
-for config in $configs; do
-    . $config_path/config.$config
+export EXPDIR=${EXPDIR:-${HOMEgfs}/parm/config}
+config_path=${EXPDIR:-${PACKAGEROOT}/gfs.${gfs_ver}/parm/config}
+for config in ${configs}; do
+    # shellcheck source-path=parm/config
+    . "${config_path}/config.${config}"
     status=$?
-    [[ $status -ne 0 ]] && exit $status
+    (( status != 0 )) && exit "${status}"
 done
 
 
 ##########################################
 # Source machine runtime environment
 ##########################################
-. $HOMEgfs/env/${machine}.env post
+# shellcheck source-path=env
+. ${HOMEgfs}/env/${machine}.env post
 status=$?
-[[ $status -ne 0 ]] && exit $status
+(( status != 0 )) && exit "${status}"
 
 
 ##############################################
 # Obtain unique process id (pid) and make temp directory
 ##############################################
 export pid=${pid:-$$}
-export outid=${outid:-"LL$job"}
+export outid=${outid:-"LL${job}"}
 export DATA=${DATA:-${DATAROOT}/${jobid:?}}
-mkdir -p $DATA
-cd $DATA
+mkdir -p "${DATA}"
+cd "${DATA}" || exit 1
 
 
 ##############################################
 # Run setpdy and initialize PDY variables
 ##############################################
-export cycle="t${cyc}z"
+export cycle="t${cyc:?}z"
 setpdy.sh
+# shellcheck disable=SC1091
```

Comment:
I am not sure littering the code w/ this type of stuff is a good idea.
Its distracting for one, and its hard to know what are we disabling and why specifically here.